Defra.Cdp.Backend.Api/Services/Deployments/DeploymentsService.cs

```diff
@@ -371,48 +376,70 @@ public async Task<Paginated<DeploymentOrMigration>> FindLatestWithMigrations(str
         if (!string.IsNullOrWhiteSpace(status))
         {
             var statusLower = status.ToLower();
+
+            // Filter deployments by status
             var statusFilter = builder.Where(d =>
                 d.Status.ToLower() == statusLower || d.Status.ToLower().Contains(statusLower));
             filter &= statusFilter;
-            // Don't filter migration status as they're completely different to deployment status.
+
+            // Filter migrations by status
+            var migrationStatusFilter = builderMigration.Where(m =>
+                m.Status.ToLower() == statusLower || m.Status.ToLower().Contains(statusLower));
+            migrationFilter &= migrationStatusFilter;
         }
 
-        var migrationCollection = Collection.Database.GetCollection<DatabaseMigration>("migrations");
+        var results = new List<DeploymentOrMigration>();
+
+        // Prepare Deployments pipeline
+        if (kind is null or Deployment)
+        {
+            var pipeline = new EmptyPipelineDefinition<Deployment>()
+                .Match(filter)
+                .Project(d => new DeploymentOrMigration { Deployment = d, Updated = d.Updated });
+
+            var deploymentResults = await Collection.Aggregate(pipeline).ToListAsync(ct);
+            results.AddRange(deploymentResults);
+        }
+
+        // Prepare Migrations pipeline
+        if (kind is null or Migration)
+        {
+            var migrationCollection = Collection.Database.GetCollection<DatabaseMigration>("migrations");
         var migrationPipeline = new EmptyPipelineDefinition<DatabaseMigration>()
             .Match(migrationFilter)
-            .Project(m => new DeploymentOrMigration
-        {
-            Migration = m,
-            Updated = m.Updated
-        });
+            .Project(m => new DeploymentOrMigration { Migration = m, Updated = m.Updated });
 
-        var pipeline = new EmptyPipelineDefinition<Deployment>()
-            .Match(filter)
-            .Project(d => new DeploymentOrMigration { Deployment = d, Updated = d.Updated})
-            .UnionWith(migrationCollection, migrationPipeline)
-            .Sort(new SortDefinitionBuilder<DeploymentOrMigration>().Descending(d => d.Updated))
+        var migrationResults = await migrationCollection.Aggregate(migrationPipeline).ToListAsync(ct);
+        results.AddRange(migrationResults);
+        }
+
+        // Sort and paginate combined results
+        results = results
+            .OrderByDescending(r => r.Updated)
             .Skip(offset + size * (page - DefaultPage))
-            .Limit(size);
-        
-        var deployments = await Collection.Aggregate(pipeline).ToListAsync(ct);
+            .Take(size)
+            .ToList();
```

Comment:
We probably want to do the pagination in the pipeline rather than serializing it to a list and then throwing most of it away. Probably wont scale too well as the data grows.

Reply:
I've reverted this work. Could do with your eyes on this in the morning if you have capacity? Would be interested in a pair on it